packages/google_sign_in/ios/Classes/GoogleSignInPlugin.m

```diff
@@ -51,16 +51,12 @@ - (instancetype)init {
 
 - (void)handleMethodCall:(FlutterMethodCall *)call result:(FlutterResult)result {
   if ([call.method isEqualToString:@"init"]) {
-    NSError *error;
-    NSString *clientId = call.arguments[@"clientId"];
-    if (clientId && ![clientId isEqual:[NSNull null]]) {
-      [GIDSignIn sharedInstance].clientID = call.arguments[@"clientId"];
-    } else {
-      [[GGLContext sharedInstance] configureWithError:&error];
-    }
+    NSString *path = [[NSBundle mainBundle] pathForResource:@"GoogleService-Info" ofType:@"plist"];
+    NSMutableDictionary *plist = [[NSMutableDictionary alloc] initWithContentsOfFile:path];
+    [GIDSignIn sharedInstance].clientID = plist[@"CLIENT_ID"];
```

Comment:
Add comment to indicate where this const is coming from. If some time in the future this breaks due to a key rename, we know where to look to fix it.